qiskit/pulse/schedule.py

```diff
@@ -142,23 +131,29 @@ def ch_duration(self, *channels: List[Channel]) -> int:
         Args:
             *channels: Channels within ``self`` to include.
         """
-        return self.timeslots.ch_duration(*channels)
+        return self.ch_stop_time(channels)
 
     def ch_start_time(self, *channels: List[Channel]) -> int:
         """Return the time of the start of the first instruction over the supplied channels.
 
         Args:
             *channels: Channels within ``self`` to include.
         """
-        return self.timeslots.ch_start_time(*channels)
+        chan_intervals = [self._timeslots[chan] for chan in channels if chan in self._timeslots]
```

Comment:
This is creating an intermediate list would it be possible to use a generator expression instead?